File: python/linked_list/test_linked_list.py
import unittest
from linked_list import *

class LinkedListTest(unittest.TestCase):
    def setUp(self):
        self.e1 = Element(1)
        self.e2 = Element(2)
        self.e3 = Element(3)
        self.e4 = Element(4)

        self.ll = LinkedList(self.e1)
        self.ll.append(self.e2)
        self.ll.append(self.e3)

    def test_get_position(self):
        self.assertEqual(self.ll.head.next.next.value, 3)
        self.assertEqual(self.ll.get_position(1).value, 1)
        self.assertEqual(self.ll.get_position(2).value, 2)
        self.assertEqual(self.ll.get_position(3).value, 3)

    def test_insert(self):
        self.ll.insert(self.e4,3)
        self.assertEqual(self.ll.get_position(3).value, 4)
        self.ll.insert(self.e4,1)
        self.assertEqual(self.ll.get_position(1).value, 4)

    def test_delete(self):
        self.ll.insert(self.e4,3)
        self.ll.delete(1)
        self.assertEqual(self.ll.get_position(1).value, 2)
        self.assertEqual(self.ll.get_position(2).value, 4)
        self.assertEqual(self.ll.get_position(3).value, 3)

if __name__ == '__main__':
    unittest.main()

File: ruby/linked_list/test_linked_list.rb
require_relative 'linked_list'
require 'test/unit'

class TestLinkedList < Test::Unit::TestCase
  def setup
    e1 = Element.new(1)
    e2 = Element.new(2)
    e3 = Element.new(3)
    @e4 = Element.new(4)

    @ll = LinkedList.new(e1)
    @ll.append(e2)
    @ll.append(e3)
  end

  def test_get_position
    assert_equal(3, @ll.head.next.next.value)
    assert_equal(3, @ll.get_position(3).value)
  end

  def test_insert
    @ll.insert(@e4,3)
    assert_equal(4, @ll.get_position(3).value)
    assert_equal(1, @ll.get_position(1).value)
    @ll.insert(@e4,1)
    assert_equal(4, @ll.get_position(1).value)
  end

  def test_delete
    @ll.insert(@e4,3)
    @ll.delete(1)
    assert_equal(2, @ll.get_position(1).value)
    assert_equal(4, @ll.get_position(2).value)
    assert_equal(3, @ll.get_position(3).value)
    @ll.delete(4)
    assert_equal(3, @ll.get_position(2).value)
  end
end
